tensorflow_addons/metrics/hamming.py

```diff
@@ -127,7 +136,15 @@ def hamming_loss_fn(y_true, y_pred, threshold, mode):
 class HammingLoss(MeanMetricWrapper):
     """Computes hamming loss."""
 
-    def __init__(self, mode, name="hamming_loss", threshold=None, dtype=tf.float32):
+    @typechecked
+    def __init__(
+        self,
+        mode: str,
+        name: str = "hamming_loss",
+        threshold: Optional[FloatTensorLike] = None,
+        dtype: Optional[FloatTensorLike] = None,
```

Comment:
We need to find the base class for tensorflow types. The common class between tf.float32 tf.int64... Etc

Reply:
Thanks @gabrieldemarmiesse.  Do you suggest using `tf.dtypes.DType`? I've tried several ways, but still have the issue. What would you suggest to do?

Reply:
Followed the https://github.com/tensorflow/addons/pull/1026 fixed the issue.